trantor/net/TcpConnection.h

```diff
@@ -58,20 +56,49 @@ class TcpConnection
     virtual void shutdown() = 0;
     virtual void forceClose() = 0;
     virtual EventLoop *getLoop() = 0;
+    template <typename T>
+    void setContext(T &&context)
+    {
+        typedef typename std::remove_cv<
+            typename std::remove_reference<T>::type>::type ContextType;
+        _contextPtr = std::make_shared<ContextType>(std::forward<T>(context));
+    }
+    template <int N>
+    void setContext(const char (&str)[N])
```

Comment:
On a more programmatic and less build-system note, although this is obviously to make taking a statically-sized string a little easier, it seems a bit special-case'y, as above I'd opt to just accept a `std::shared_ptr` of any internal type instead and let the user construct it as they wish.